### ISSUE TITLE
Current Temperature FGLair

### ISSUE THREAD

**Issue by kevtuning** [deiger/AirCon#255]
Hi,

If I understand well , the only way to have the current temperature for FGLair AC (Fujitsu, ...) when running as an add-on is to integrate the Pull requests #222 and #224.

Is it correct ? and if yes, @deiger is it possible to integrate it in the next release ?

Thank you very much,

**Comment by legantois** [comment]
Same request here #240 

I believe it is a key enhancement !

2x thank you very much

**Comment by kevtuning** [comment]
Any chance to get it merged with 2023.10 ?

**Comment by legantois** [comment]
Is there any way to get all integrated ?

**Comment by legantois** [comment]
Is there any way to get all integrated ?

**Comment by legantois** [comment]
seems we will never get this new feature on the official release

Could someone explains how to install an addon PR ? 

my config:
Core 2024.2.3
Supervisor 2024.02.0
Operating System 11.5
Frontend 20240207.1

**Comment by xury77** [comment]
Through lot of  trials and errors, I was able to clone the repo and create a docker image from PR #224. I don't see the temperature on the thermostat card but mqtt sends the correct data as in the FGLair application. Link to paste in HA for installation: https://github.com/xury77/AirCon.git

**Comment by kevtuning** [comment]
> I don't see the temperature on the thermostat card but mqtt sends the correct data as in the FGLair application. Link to paste in HA for installation: https://github.com/xury77/AirCon.git

Hi @xury77, what's the mqtt topic ? 

Is it well 'display_temperature' ?

**Comment by kevtuning** [comment]
I don't really see in the sources where it is doing the mapping display_temperature to current_temperature

**Comment by xury77** [comment]
I think is some `env_temp` related with `f_temp_in`.

**Comment by kevtuning** [comment]
@xury77 

I am not an expert but I would try to test to change the line 247 of main.py

 config['display_temperature_state_topic'] = mqtt_topics['pub'].format(device.mac_address,
                                                                      topics['display_temperature'])
                                                                      
 by 

 config['current_temperature_topic'] = mqtt_topics['pub'].format(device.mac_address,
                                                                      topics['display_temperature'])

**Comment by xury77** [comment]
Yes. According https://www.home-assistant.io/integrations/climate.mqtt/

> current_temperature_topic string (optional)
> The MQTT topic on which to listen for the current temperature. A "None" value received will reset the current temperature. Empty values (''') will be ignored.
I will check it.

**Comment by xury77** [comment]
Yes at least it works: 
![Zrzut ekranu z 2024-07-25 17-05-54](https://github.com/user-attachments/assets/b190ee64-771a-4b19-b352-36669ba29f5a)

**Comment by xury77** [comment]
New release 0.3.20 pushed to dockerhub and available at https://github.com/xury77/AirCon At least current temperature is displayed as above.

**Comment by legantois** [comment]
works on my side two

![image](https://github.com/user-attachments/assets/708c9233-4c82-48d6-a27f-9f6e7485f278)


Many Thanks for the work !!!

**Comment by xury77** [comment]
I think I can be able to add some features for FGlair devices such as: vertical positions, eco, boost mode, energy saving, outdoor noise  etc: I'm not a python developer but I like doing some hacks :)

**Comment by jonad0521** [comment]
this is great! i've switched to your fork. any other features you want to add are welcome for me!
